Twitch/Twitch.Base/Services/NewAPI/UsersService.cs

```diff
@@ -99,6 +99,29 @@ public async Task<IEnumerable<UserModel>> GetUsers(IEnumerable<string> userIDs,
             return await this.GetDataResultAsync<UserModel>("users?" + string.Join("&", parameters));
         }
 
+        /// <summary>
+        /// Gets number of followers for and/or to a user.
+        /// </summary>
+        /// <param name="from">The user to search for who they follow</param>
+        /// <param name="to">The user to search for who follows them</param>
+        /// <returns>The total number of followers</returns>
+        public async Task<long> GetFollowsCount(UserModel from = null, UserModel to = null)
```

Comment:
This one doesn't quite make sense with the intention of the method and the parameters. The purpose of the original GetFollows() method was to cover one of three scenarios:
- Get the people user A is following
- Get the followers for user B
- See if user A is following user B

Since this is specifically focused on getting a follow count, the real use case is for seeing how many people are following a specific channel. So we don't need the "from" and "to" in this case, we just need a single user who we are checking the follow count against.

Reply:
Understood.  I didn't want to make assumptions on how a person might use the library, therefore I thought the best option would be to follow the existing pattern for getting users but instead of returning the actual items just show the count.  This allows for the most flexibility for people using the library.

Reply:
Is there a reason to limit the use to just getting the follow count for a specific person / channel?

Reply:
It's less about limiting the user and more is there a reasonable use-case for providing the option. If you can make an API call simpler, there's really no reason not too. There's 3 scenarios of usage for the GetFollows() API call, however only 2 of those scenario would apply to getting a follow count:

- Get the people user A is following (IE, how many followers does this user have?)
- Get the followers for user B (IE, how many people is this user following?)

The 3rd scenario doesn't make sense when it comes to follow count, as it's checking if user A is following user B. So only at most one record is going to be returned (either they're following or not following), so the count will always be either 0 or 1. There's just not really much of a point for that scenario, so presenting an option where you can simply either or both of the parameters doesn't make much sense.

It would be better to make two separate methods, one for getting the **follower** count (scenario 1) and one for get the **follow** count (scenario 2). However, when realistically would there be a scenario where someone needs to know the number of people they are following? There could be a case, but it certainly wouldn't be a common case. So scenario 2 makes less sense to expose, leaving just scenario 1.

There's also just code maintenance; the more methods you add, the more that need to be maintained. If no one is using a method, then it's just dead code sitting there. So it's better to focus on the things that are needed first and then see if there's actually a need for the other things later.

Based on the above, I would also change this to **GetFollowerCount** instead to better indicate it for getting the followers for a channel (and not to confuse it with getting the follows this channel has).